The job **`x86_64-gnu-aux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/15093244188/job/42424570099) [(plain)](https://github.com/rust-lang-ci/rust/commit/1d4757cd30d7240b2338cb05c7548b2df336371a/checks/42424570099/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
test fs::tests::canonicalize_works_simple ... ok
test fs::tests::chmod_works ... FAILED
test fs::tests::copy_file_does_not_exist ... ok
test fs::tests::copy_file_dst_dir ... ok
error: unsupported operation: can't call foreign function `fchmod` on OS `linux`
##[error]    --> library/std/src/sys/fs/unix.rs:1476:27
     |
1476 |         cvt_r(|| unsafe { libc::fchmod(self.as_raw_fd(), perm.mode) })?;
     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `fchmod` on OS `linux`
     |
     = help: if this is a basic API commonly used on this target, please report an issue with Miri
     = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
     = note: BACKTRACE on thread `fs::tests::copy`:
     = note: inside closure at library/std/src/sys/fs/unix.rs:1476:27: 1476:68
note: inside `sys::pal::unix::cvt_r::<i32, {closure@library/std/src/sys/fs/unix.rs:1476:15: 1476:17}>`
    --> library/std/src/sys/pal/unix/mod.rs:317:19
     |
317  |         match cvt(f()) {
     |                   ^^^
note: inside `sys::fs::unix::File::set_permissions`
    --> library/std/src/sys/fs/unix.rs:1476:9
     |
1476 |         cvt_r(|| unsafe { libc::fchmod(self.as_raw_fd(), perm.mode) })?;
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `fs::File::set_permissions`
    --> library/std/src/fs.rs:1072:9
     |
1072 |         self.inner.set_permissions(perm.0)
---
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `sys::fs::unix::copy`
    --> library/std/src/sys/fs/unix.rs:2044:37
     |
2044 |     let (writer, writer_metadata) = open_to_and_set_permissions(to, &reader_metadata)?;
     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `sys::fs::copy`
    --> library/std/src/sys/fs/mod.rs:118:12
     |
118  |     return imp::copy(from, to);
     |            ^^^^^^^^^^^^^^^^^^^
note: inside `fs::copy::<&path::PathBuf, &path::PathBuf>`
    --> library/std/src/fs.rs:2604:5
     |
2604 |     fs_imp::copy(from.as_ref(), to.as_ref())
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `fs::tests::copy_file_dst_exists`
    --> library/std/src/fs/tests.rs:956:12
     |
956  |     check!(fs::copy(&input, &output));
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside closure
    --> library/std/src/fs/tests.rs:949:26
     |
948  | #[test]
     | ------- in this procedural macro expansion
949  | fn copy_file_dst_exists() {
     |                          ^
     |
    ::: /checkout/library/core/src/macros/mod.rs:1656:5
     |
1656 |     pub macro test($item:item) {
     |     -------------- in this expansion of `#[test]`
note: inside `<{closure@library/std/src/fs/tests.rs:949:1: 961:2} as core::ops::FnOnce<()>>::call_once - shim`
    --> /checkout/library/core/src/ops/function.rs:250:5
     |
250  |     extern "rust-call" fn call_once(self, args: Args) -> Self::Output;
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `<fn() -> core::result::Result<(), realstd::string::String> as core::ops::FnOnce<()>>::call_once - shim(fn() -> core::result::Result<(), realstd::string::String>)`
    --> /checkout/library/core/src/ops/function.rs:250:5
     |
250  |     extern "rust-call" fn call_once(self, args: Args) -> Self::Output;
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

error: test failed, to rerun pass `-p std --lib`

Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/cargo-miri runner /checkout/obj/build/x86_64-unknown-linux-gnu/stage1-std/miri/x86_64-unknown-linux-gnu/debug/deps/std-67f09f76e8983246 -Z unstable-options --format json` (exit status: 1)
note: test exited abnormally; to see the full output pass --nocapture to the harness.
Build completed unsuccessfully in 0:01:10
make: *** [Makefile:58: check-aux] Error 1
  local time: Sun May 18 08:08:13 UTC 2025
  network time: Sun, 18 May 2025 08:08:13 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
```

</details>

--- a/src/bootstrap/mk/Makefile.in
+++ b/src/bootstrap/mk/Makefile.in
@@ -60,7 +60,8 @@ check-aux:
 		library/alloc \
 		$(BOOTSTRAP_ARGS) \
 		--no-doc
-	# Some doctests use file system operations to demonstrate dealing with `Result`.
+	# Some doctests use file system operations to demonstrate dealing with `Result`,
+	# so we have to run them with isolation disabled.
 	$(Q)MIRIFLAGS="-Zmiri-disable-isolation" \
 		$(BOOTSTRAP) miri --stage 2 \
 		library/coretests \
@@ -70,22 +71,19 @@ check-aux:
 		--doc
 	# In `std` we cannot test everything, so we skip some modules.
 	$(Q)MIRIFLAGS="-Zmiri-disable-isolation" \
-		$(BOOTSTRAP) miri --stage 2 library/std \
+		$(BOOTSTRAP) miri --stage 2 \
+		library/std \
 		$(BOOTSTRAP_ARGS) \
-		--no-doc -- \
-		--skip fs:: --skip net:: --skip process:: --skip sys::fd:: --skip sys::pal::
-	$(Q)MIRIFLAGS="-Zmiri-disable-isolation" \
-		$(BOOTSTRAP) miri --stage 2 library/std \
-		$(BOOTSTRAP_ARGS) \
-		--doc -- \
+		-- \
 		--skip fs:: --skip net:: --skip process:: --skip sys::fd:: --skip sys::pal::
 	# Also test some very target-specific modules on other targets
 	# (making sure to cover an i686 target as well).
 	$(Q)MIRIFLAGS="-Zmiri-disable-isolation" BOOTSTRAP_SKIP_TARGET_SANITY=1 \
-		$(BOOTSTRAP) miri --stage 2 library/std \
-		$(BOOTSTRAP_ARGS) \
+		$(BOOTSTRAP) miri --stage 2 \
+		library/std \
 		--target aarch64-apple-darwin,i686-pc-windows-msvc \
-		--no-doc -- \
+		$(BOOTSTRAP_ARGS) \
+		-- \
 		time:: sync:: thread:: env::
 dist:
 	$(Q)$(BOOTSTRAP) dist $(BOOTSTRAP_ARGS)

--- a/src/bootstrap/mk/Makefile.in
+++ b/src/bootstrap/mk/Makefile.in
@@ -54,7 +54,8 @@ check-aux:
 		src/etc/test-float-parse \
 		$(BOOTSTRAP_ARGS)
 	# Run standard library tests in Miri.
-	$(Q)$(BOOTSTRAP) miri --stage 2 \
+	$(Q)MIRIFLAGS="-Zmiri-strict-provenance" \
+		$(BOOTSTRAP) miri --stage 2 \
 		library/coretests \
 		library/alloctests \
 		library/alloc \

